docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,416 @@
+type CommunicationLayer interface {
+    // A process `s` invokes this function to send a datagram `g` to
+    // a destination process `d`.
+    Send(g Datagram, d Process) null
+
+    // This callback function triggers at process `d` to deliver datagram `g`
+    // that process `s` previously sent.
+    Deliver(g Datagram, s Process) null
+}
+```
+
+- **Expected precondition:**
+    - the datagram `g` is non-null;
+    - the destination `d` and source `s` processes are non-null;
+- **Expected postcondition:** none
+- **Error condition:** none
+
+- **Properties of the communication layer:**
+    - **[CH/CommAssumption/1/Consistency]** If some process `d` triggers `Deliver(g, s)`, then process `s` previously invoked `Send(g, d)`.
+    - **[CH/CommAssumption/2/Termination]** If some process `s` invokes `Send(g, d)`, then eventually the callback`Deliver(g, s)` triggers at process `d`.
+
+Neither of the `Send` or `Deliver` are blocking, and they have no return value.
+The communication layer is _asynchronous_: There is no guarantee on how much time it will take for the destination of a datagram to deliver that datagram.
+
+
+###### Fault assumptions
+Processes `I` and `R` may be subject to arbitrary (Byzantine) faults.
+The properties we define below are guaranteed to hold iff both `I` and `R` behave correctly, i.e., do not deviate from any protocol we describe below. .
+Beside these two processes, the system comprises an arbitrary number of other chains, any of which may be Byzantine; these other chains may, for instance, try to impersonate `I` or `R`.
+
+
+###### Additional remarks on the system model
+We explicitly overlook here two assumptions of synchrony that may be necessary for achieving liveness of **other** protocols, but they are not necessary within the connection handshake protocol.
+These assumptions are: (i) partial synchrony is necessary for achieving liveness of a chain if that chain is implemented using a deterministic consensus protocol (e.g., Tendermint Core); (ii) synchrony is necessary for guaranteeing message delivery in any connection-oriented protocol such as TCP, which is most likely the underlying communication layer running between `I` and `R`.
+
+
+#### Properties
+
+###### Preliminary definitions
+We introduce several abstractions that are relevant at this point:
+
+1. A `ConnectionParty` is a data type encapsulating the identifier(s) of a party (either initiator or responder process) involved in a connection.
+We can think of `ConnectionParty` as a concrete way of specifying a process in the CH protocol.
+
+```golang
+type ConnectionParty interface {
+    ConnectionIdentifier    Identifier
+    ClientIdentifier        Identifier
+    Prefix                  Prefix // TODO -- keep this?
+}
+```
+
+2. A `ConnectionState` captures the state of a connection, and may have one of the following values. [TODO: not valid Go code below, keep?]
+
+```typescript
+enum ConnectionState {
+    INIT,
+    TRYOPEN,
+    OPEN,
+}
+```
```

Comment:
As far as I know, nope.
Note that I'm using [this](https://github.com/cosmos/ics/tree/master/spec/ics-003-connection-semantics#data-structures) as a guide. Let me know otherwise!